src/Rings/groebner.jl

```diff
@@ -1221,3 +1232,104 @@ function _compute_groebner_basis_using_fglm(I::MPolyIdeal,
 	dim(I) == 0 || error("Dimension of ideal must be zero.")
 	I.gb[destination_ordering] = _fglm(G, destination_ordering)
 end
+
+
+# TODO: does the target ordering have to be global?
+@doc Markdown.doc"""
+    _groebner_basis_with_hilbert(I::MPolyIdeal,
+                                 destination_ordering::MonomialOrdering;
+                                 weights::Vector{E} = ones(ngens(base_ring(I))),
+                                 complete_reduction::Bool = false) where {E <: Integer}
+
+**Note**: Internal function, subject to change, do not use.
+
+Compute a Gröbner basis of `I` with respect to `destination_ordering`
+using a Hilbert Series driven method as follows: If a Gröbner basis
+for `I` is present, compute the Hilbert series of `I` and use it to
+optimize the Gröbner basis computation for `I`
+w.r.t. `destination_ordering`. If no Gröbner basis for `I` is present
+compute the Hilbert series for `I` if the base field of `I` has
+positive characteristic, otherwise compute the Hilbert series for `I`
+modulo a randomly chosen prime. Use the resulting Hilbert series to
+optimize the Gröbner basis computation for `I`
+w.r.t. `target_ordering`.
+
+`I` must be given by generators homogeneous w.r.t. `weights`.
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"]);
+
+julia> I = ideal(R, [x^2 + y*z, x*y - y*z]);
+
+julia> Oscar._standard_basis_with_hilbert(I, lex(R), complete_reduction = true)
+Gröbner basis with elements
+1 -> y^2*z + y*z^2
+2 -> x*y - y*z
+3 -> x^2 + y*z
+with respect to the ordering
+lex([x, y, z])
+```
+"""
+
+function _standard_basis_with_hilbert(I::MPolyIdeal,
+                                      destination_ordering::MonomialOrdering;
+                                      weights::Vector{E} = ones(Int32, ngens(base_ring(I))),
+                                      complete_reduction::Bool = false) where {E <: Integer}
+  
+  all(p -> _is_homogeneous_weights(p, weights), gens(I)) || error("I must be given by generators homogeneous with respect to given weights.")
+	isa(coefficient_ring(base_ring(I)), AbstractAlgebra.Field) || error("The underlying coefficient ring of I must be a field.")
+	haskey(I.gb, destination_ordering) && return I.gb[destination_ordering]
+  if isempty(I.gb) && iszero(characteristic(base_ring(I)))  
+    while true
+      p = 0
+      while true
+        p = rand(2^15:2^16)
+        if isprime(p)
+          break
```

Comment:
```suggestion
    r = rand(2^15:2^16)
    p = next_prime(r)
```

Maybe that's more efficient?